src/basic/exec-util.h

```diff
@@ -17,6 +17,23 @@
   along with systemd; If not, see <http://www.gnu.org/licenses/>.
 ***/
 
+#include <stdbool.h>
+
 #include "time-util.h"
 
-void execute_directories(const char* const* directories, usec_t timeout, char *argv[]);
+typedef int (*gather_stdout_callback_t) (int fd, void *arg);
```

Comment:
For most other cases where we have a callback that takes arbitrary userdata ptrs we called that ptr `userdata`. I think we should do that here, too!